istioctl/pkg/multixds/gather.go

```diff
@@ -389,7 +389,15 @@ func mapShards(responses []*discovery.DiscoveryResponse) (map[string]*discovery.
 	retval := map[string]*discovery.DiscoveryResponse{}
 
 	for _, response := range responses {
-		retval[CpInfo(response).ID] = response
+		id := CpInfo(response).ID
+		if prevResp, ok := retval[id]; ok && id == "" {
+			// In some special environment (e.g., managed control plane),
```

Comment:
Why don't we just tell these folks to set the variable

Reply:
hmm, I see. Will check the option first.